lib/spack/spack/filesystem_view.py

```diff
@@ -96,6 +97,8 @@ def view_copy(src, dst, view, spec=None):
                 files=[dst],
                 prefixes=prefix_to_projection
             )
+        stat = os.stat(src)
+        os.chown(dst, stat.st_uid, stat.st_gid)
```

Comment:
Can you put a try/except block around this? This will fail without root permissions if you are not the original owner of the file or don't belong to the group it originally belonged to. I think we should be best-effort about this instead.

Reply:
Done